### PR TITLE
(HTCONDOR-2505)  Implement `htcondor credential listall`.

### PR DESCRIPTION
**Pull request by Todd-L-Miller**
Presently only for OAuth2.

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: docs/version-history/feature-versions-23-x.rst [hunk 1]
```diff
@@ -22,6 +22,10 @@ New Features:
   a job.
   :jira:`2510`
 
+- Added a new verb to ``htcondor credential``, ``listall``, which allows the
```
Comment (ColeBollig) [inline docs/version-history/feature-versions-23-x.rst:25]:
```suggestion
- Added a new verb to :tool:`htcondor credential listall`, which allows the
```
FYI: the above works to create a reference to the `htcondor` tool manual

Comment (ColeBollig) [reply docs/version-history/feature-versions-23-x.rst:25]: Note you could also keep this similar to what you have as such:
:tool:`htcondor credential`, ``listall``

File: src/condor_tools/htcondor_cli/credential.py [hunk 2]
```diff
@@ -10,6 +10,129 @@
...
+
+                for file in entry.iterdir():
+                    if file.suffix != ".top":
+                        continue
+                    credentials_by_user[entry.name][file.stem] = {
+                        'refresh':  int(file.stat().st_mtime),
+                        'jobs':     defaultdict(list),
+                    }
+        except PermissionError:
+            print("You don't have permission to read the credentials directory.")
```
Comment (jasoncpatton) [inline src/condor_tools/htcondor_cli/credential.py:82]: Could/should this message include the owner of the creddir?

Comment (Todd-L-Miller) [reply src/condor_tools/htcondor_cli/credential.py:82]: The owner of the creddir has been `condor.condor` on every machine I've checked so far (with the group sticky bit set, I guess?), but I'm pretty sure we don't want to tell people to su(do) to the condor user, so I'd rather not get into that.  I'd be fine with adding something like, "Try again as the root or admnistrator account".

File: docs/version-history/feature-versions-23-x.rst (hunk 1 above)
Comment (Todd-L-Miller) [reply docs/version-history/feature-versions-23-x.rst]: The second form makes more sense to me; commit pushed.

**Pull request merged**